backend/services/agentops.py

```
@@ -0,0 +1,748 @@
+                        self.span.set_attribute(SpanAttributes.LLM_USAGE_TOTAL_TOKENS, total)
+                
+                # Additional usage attributes
+                if hasattr(response.usage, "reasoning_tokens") and response.usage.reasoning_tokens:
+                    self.span.set_attribute(SpanAttributes.LLM_USAGE_REASONING_TOKENS, response.usage.reasoning_tokens)
+                if hasattr(response.usage, "cache_creation_input_tokens") and response.usage.cache_creation_input_tokens:
+                    self.span.set_attribute(SpanAttributes.LLM_USAGE_CACHE_CREATION_INPUT_TOKENS, response.usage.cache_creation_input_tokens)
+                if hasattr(response.usage, "cache_read_input_tokens") and response.usage.cache_read_input_tokens:
+                    self.span.set_attribute(SpanAttributes.LLM_USAGE_CACHE_READ_INPUT_TOKENS, response.usage.cache_read_input_tokens)
+            
+            
+            # Set successful status
```

Flake8 (7.2.0)

[error] too many blank lines (2)

(E303)